### ISSUE TITLE
Odd feature flags on enable_io

### ISSUE THREAD

**Issue by Darksonn** [tokio-rs/tokio#2973]
The feature flags on [`enable_io`][1] are:

> This is supported on crate feature rt and (crate feature net, or crate feature process, or Unix and crate feature signal) only.

This may be the exact situation where IO is available, but I think we should do something to simplify this. We can enable it in more situations than exactly where it does something without issue.

[1]: https://docs.rs/tokio/0.3.0/tokio/runtime/struct.Builder.html#method.enable_io

**Comment by carllerche** [comment]
Do you have a proposed change? I don't think this would require a breaking change and we can remove the 1.0 tag?

**Comment by Darksonn** [comment]
Yeah, my proposal would be to increase the cases in which it is available, which is not breaking.

**Comment by rifuki** [comment]
Proposed change: move `enable_io` and `max_io_events_per_tick` from `cfg_io_driver!` to `cfg_rt!`, so they exist whenever `rt` is enabled.

The gate isn't load-bearing. The fields it guards are already unconditional — `enable_io` and `nevents` on `Builder` (builder.rs:63-64) carry no cfg — and when no I/O driver is compiled in, `create_io_stack` ignores the flag entirely, so setting it is a no-op rather than an error. `enable_all` (builder.rs:396) already relies on exactly that: the method is ungated and only the inner `self.enable_io()` call carries the `net`/`process`/`signal` cfg. The result today is that under bare `rt` you can call `enable_all()` but `enable_io()` does not exist.

This only widens where the methods are available, so it isn't breaking, and it reduces the rendered feature line to `rt`. #8308 tried to reach the same place by editing the `doc(cfg)` text alone, which is how it ended up claiming `process` works on Windows; changing the gate itself makes the simpler line true instead of just shorter.

Patch is one line and verified locally: #8387.

**Comment by Darksonn** [comment]
The runtime builder in its entirety is already gated under `cfg_rt!`, so that'd be equivalent to not having it under any gate at all.

**Comment by rifuki** [comment]
You're right, thanks — `mod builder` is inside `cfg_rt!` in `runtime/mod.rs`, so `cfg_rt!` there gates nothing.

I've pushed a version that removes the gate instead of replacing it: the two methods move into the main `impl Builder` block. Same effect, without claiming a gate that does nothing.

The behaviour change is unchanged either way — under bare `rt`, `enable_io()` currently does not exist while `enable_all()` compiles and quietly skips I/O.
